src/main/java/frc/robot/commands/Shoot.java

```diff
@@ -66,19 +96,38 @@ public void initialize() {
                 // Reset the timer
                 shootTimer.reset();
                 shootTimer.stop();
-                shooter.resetPID();
-                drive.setIsAlign(true); // Enable alignment mode on the drivetrain
-                drive.onlyUseTags(new int[]{3, 4, 7, 8});
+
+                // Enable alignment mode on the drivetrain
+                drive.setIsAlign(true);
+
+                // only use the April Tags on the speakers
+                // April Tags 3, 4, 7, and 8 are the speaker tags on the Alliance Wall.
+                // this is a backdoor to vision...
+                drive.onlyUseTags(aprilTagsSpeaker);
+
+                // we are not shooting yet
                 shooting = false;
+
+                // the alliance determines which speaker we target
+                // the speakerPose need not be calculated every execute; do it in initialize()
+                speakerPose = (Robot.getAlliance() == Alliance.Red) ?
+                                VisionConstants.RED_SPEAKER_POSE : VisionConstants.BLUE_SPEAKER_POSE;
         }
+
         @Override
         public void execute() {
+                // TODO: positive x displacement -> left of speaker only for Blue Alliance
                 // Positive x displacement means we are to the left of the speaker
+                // TODO: doesn't positive y displacement mean we are above the speaker?
                 // Positive y displacement means we are below the speaker.
```

Comment:
Left is only for red, not blue, and positive y displacement should mean we are below the speaker. Displacement should be from note start point to note end point, which is the speaker.

Reply:
Although displacement is inverted for some reason, so everything in my last comment should be inverted.